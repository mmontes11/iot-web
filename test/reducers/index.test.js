import deepFreeze from "deep-freeze";
import rootReducer from "reducers";

jest.mock("reducers/auth", () => (state = false, { type }) => {
  switch (type) {
    case "ACTION":
      return "STATE_AUTH";
    default:
      return state;
  }
});

describe("reducers/index", () => {
  it("combines other reducers", () => {
    expect(rootReducer).toMatchSnapshot();
  });

  const state = {
    isAuth: "INITIAL_AUTH"
  };

  deepFreeze(state);

  it("has a default state", () => {
    expect(rootReducer(undefined, { type: "@init" })).toMatchSnapshot();
  });
  it("reduces nothing", () => {
    expect(rootReducer(state, { type: "WHATEVER" })).toMatchSnapshot();
  });
  it("reduces other actions", () => {
    expect(rootReducer(state, { type: "ACTION" })).toMatchSnapshot();
  });
});
