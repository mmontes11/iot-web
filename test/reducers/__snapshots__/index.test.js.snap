// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/index combines other reducers 1`] = `[Function]`;

exports[`reducers/index has a default state 1`] = `
Object {
  "isAuth": false,
}
`;

exports[`reducers/index reduces nothing 1`] = `
Object {
  "isAuth": "INITIAL_AUTH",
}
`;

exports[`reducers/index reduces other actions 1`] = `
Object {
  "isAuth": "STATE_AUTH",
}
`;
